build-tools/scripts/mono-version

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash -e
+
+sysname=`uname`
+cmdname=`basename "$0"`
+if [ "$sysname" = Darwin ]; then
+	MONO_PREFIX=/Library/Frameworks/Mono.framework/Home
+	PKG_CONFIG_PATH=/Library/Frameworks/Mono.framework/Home/bin/pkg-config
+	MONO_VERSION=`"$PKG_CONFIG_PATH" --modversion mono`.`cat $MONO_PREFIX/updateinfo | cut -d' ' -f2 | cut -c6- | awk '{print(int($0))}'`
+	echo $MONO_VERSION
+	exit 0
+fi
+if [ "$sysname" = Linux ]; then
+	MONO_VERSION=`pkg-config --modversion mono`
```

Comment:
This may be confusing on systems where the developer has another Mono in `PATH` in front of the system Mono and the non-system mono `.pc` files aren't in `PKG_CONFIG_PATH` before the system ones. This may result in building X.A with mono different than the one detected.
Similarly, it would cause problems if `pkg-config` is absent but locally built Mono is present and no system Mono is installed.
I think, the check should be more extensive (possibly for macOS above too, as the remarks apply to it as well):

- use `which mono` to find the location of the "default" Mono in this shell instance
- obtain Mono version with `mono --version`
- check whether `pkg-config` exists
- use `pkg-config` to see whether the default Mono is the same as the one reported by `pkg-config`
- compare the Mono versions and if *either* of them is too old (especially the default one - that would be used to build X.A) warn (if the non-default Mono is too old) or error out (if the default Mono is too old)

Might seem a bit extensive but I believe we should be as flexible and as correct as possible.